lib/reek/tree_walker.rb

```diff
@@ -105,17 +111,19 @@ def process_default(exp)
 
     def process_args(_) end
 
-    # :reek:TooManyStatements: { max_statements: 6 }
+    # :reek:TooManyStatements: { max_statements: 8 }
     # :reek:FeatureEnvy
     def process_send(exp)
+      method_name = exp.method_name
       if exp.visibility_modifier?
-        element.track_visibility(exp.method_name, exp.arg_names)
-      end
-      if exp.attribute_writer?
+        element.track_visibility(method_name, exp.arg_names)
+      elsif exp.attribute_writer?
         exp.args.each do |arg|
           next unless arg.type == :sym
           new_context(Context::AttributeContext, arg, exp)
         end
+      else
+        new_context(Context::SendContext, exp, method_name)
```

Comment:
Not sure I follow - I need to create SendContexts so I can use them here: https://github.com/troessner/reek/pull/724/files#diff-18413dbca05d0a29829ff59ac73d71e6R22